src/memorylayout.jl

```diff
@@ -18,6 +18,8 @@ abstract type AbstractRowMajor <: AbstractDecreasingStrides end
 struct DenseRowMajor <: AbstractRowMajor end
 struct RowMajor <: AbstractRowMajor end
 struct DecreasingStrides <: AbstractIncreasingStrides end
+struct FirstMajor <: AbstractStridedLayout end
+struct SecondMajor <: AbstractStridedLayout end
```

Comment:
`FirstMajor` and `SecondMajor` should be more descriptive. Perhaps something like `ColumnMajorInFirstDimension`?

Reply:
Yes I agree... is `AlmostColumnMajor` too obscure? Doesn't generalise... `FirstDimensionMajor` +  `SecondDimensionMajor`? `UnitStrideFirst` + `UnitStrideSecond`?

Reply:
Using `Major` might be a bad idea as it conflates with `ColumnMajor`, which is more specific. 

What about `UnitStride{N}`? This would work for any `N`, including `UnitStride{1}` and `UnitStride{2}`.

Reply:
I wondered a bit about generalising to any N, it will just require a little more thought about permutations...

Reply:
I made a version with `UnitStride{dim}` now.